src/lava/lib/optimization/solvers/generic/sub_process_models.py

```diff
@@ -57,19 +57,40 @@ def __init__(self, proc):
             wta_weight
             * np.logical_not(np.eye(shape[1] if len(shape) == 2 else 0)),
         )
-        temperature = proc.hyperparameters.get("temperature", 1)
-        refract = proc.hyperparameters.get("refract", 0)
-        init_value = proc.hyperparameters.get("init_value", np.zeros(shape,
-                                                                     dtype=int))
-        init_state = proc.hyperparameters.get("init_state", np.zeros(shape,
-                                                                     dtype=int))
-
-        self.s_bit = BoltzmannAbstract(temperature=temperature,
-                                       refract=refract,
-                                       init_state=init_state,
-                                       shape=shape,
-                                       cost_diagonal=diagonal,
-                                       init_value=init_value)
+        neuron_model = proc.hyperparameters.get("neuron_model", 'nebm')
+        if neuron_model == 'nebm':

```

Comment:
Thanks for adding the option to switch between models :-)

Reply:
Yes, the option is great. From the user perspective, I would actually prefer to switch between algorithms rather than neuron models, of course these are equivalent but i'd say a user with no background in neuromorphics will find it less natural to talk about neuron model.